docs/30DaysOfPWA/advanced-capabilities/06.md

```diff
@@ -1,15 +1,174 @@
+async function publishPostOnConnected(post) {
+  const registration = await navigator.serviceWorker.ready;
+  try {
+    // First, you should write a code to save the data you want to send later. It's a good idea to have this function ready for multiple posts published while offline. Using IndexedDB is the best option for storing this kind of data.
+    await savePost(post);
+
+    // Then, the sync registration itself. We give it a name because we can have multiple syncs for various parts of the app functionality:
+    await registration.sync.register('sync-post');
+
+    // Finally, you can inform the user about it:
+    showNotification('Your post will be published automatically right after connection is restored. It is safe to close the app.');
+  } catch {
+    console.error('Background Sync registration failed');
+  }
+}
+```
+
+It's a really good idea to wrap this code by a feature detection:
+
+```javascript
+if ('serviceWorker' in navigator && 'SyncManager' in window) {
+   publishPostOnConnected(post)
+} else {
+  console.log('Background Sync is not supported');
+}
+```
+
+In the service worker, you listen and react to the `sync` event named `sync-post`:
+
+```javascript
+self.addEventListener('sync', event => {
+    if (event.tag === 'sync-post') {
+      event.waitUntil(
+
+        // You have to implement the function that iterates over the preserved posts and sends them:
+        publishSavedPosts()
+      );
+    }
+});
+```
+
+Here, `publishSavedPosts` function should return a promise indicating the success/failure of sending data. If it's rejected, the next sync will be scheduled to retry.
+
+Retrying failed requests can be automated and implemented in a declarative way by using the [Workbox Background Sync](https://developers.google.com/web/tools/workbox/modules/workbox-background-sync) module.
+
+## Periodic synchronization
+
+THe majority of the platforms have APIs that allow native applications to periodically run some tasks in the background without the user's interaction. For example, a social media app can fetch and store new posts from the server from time to time - to display a fresher feed when the user starts the app. Talking about the freshness of the application itself - mobile platforms have built-in mechanisms for that, desktop platforms allow developers to build corresponding services.
+
+What's about the web? Can we keep installed web frontend application and data it uses up to date on our users' devices? Yes! With the API called Periodic Background Sync API, we can ask the background service of the browser to run parts of our code with some periodicity regardless of the main application status (whether it's running or not). Like Background Sync, this API only provides an event to the service worker - you have to implement the actual "payload" yourself. For example, it could be a fetch to check for the updated app version and/or its content followed by corresponding resources update in the browser cache.
```

Comment:
```suggestion
What about the web? Can we keep installed web frontend application and data it uses up to date on our users' devices? Yes! With the API called Periodic Background Sync API, we can ask the background service of the browser to run parts of our code with some periodicity regardless of the main application status (whether it's running or not). Like Background Sync, this API only provides an event to the service worker - you have to implement the actual "payload" yourself. For example, it could be a fetch to check for the updated app version and/or its content followed by corresponding resources update in the browser cache.
```